manticore/models/linux.py

```diff
@@ -1485,20 +1485,13 @@ def syscall(self):
 
                 }
 
-        index, arguments, writeResult  = self.current.get_syscall_description()
+        index = self.current.ABI.syscall_number()
 
         if index not in syscalls:
             raise SyscallNotImplemented(64, index)
 
-        func = syscalls[index]
+        return self.current.ABI.invoke_syscall(syscalls[index])
```

Comment:
Wonder if it's better to decoreate all sys_xxxxx function methods with this so when you browse the class you can detect system calls implementations without relying in the sys_xxx name.